src/slicer.cpp

```diff
@@ -189,28 +189,32 @@ SlicerLayer::findPossibleStitches(
     // maximum distance squared
     int64_t max_dist2 = max_dist * max_dist;
 
-    // Represents an end point of a polyline in open_polylines.
-    // polyline_end == open_polylines[polyline_idx].back()
+    // Represents a terminal point of a polyline in open_polylines.
     struct StitchGridVal
     {
         unsigned int polyline_idx;
-        Point polyline_end;
+        // Depending on the SparseGrid, either the start point or the
+        // end point of the polyline
+        Point polyline_term_pt;
     };
 
     struct StitchGridValPointAccess
     {
         Point operator()(const StitchGridVal &val) const
         {
-            return val.polyline_end;
+            return val.polyline_term_pt;
         }
     };
 
     // Used to find nearby end points within a fixed maximum radius
-    SparseGridInvasive<StitchGridVal,StitchGridValPointAccess> grid(cell_size);
+    SparseGridInvasive<StitchGridVal,StitchGridValPointAccess> grid_ends(cell_size);
```

Comment:
Shouldn't this now just be a `SparseGrid` rather than a `SparseGridInvasive`? You've now changed the element type to include the point, which is precisely what SparseGrid does.